backend/Controllers/RecipeController.js

```diff
@@ -647,3 +647,58 @@ const RecipeController = {
 };
 
 export default RecipeController;
+
+export const getRecipeOfTheDay = async (req, res) => {
+  try {
+    const today = new Date();
+    today.setHours(0, 0, 0, 0);
+
+    let recipeOfTheDay = await Recipe.findOne({
+      featured: true,
+      lastFeatured: { $gte: today }
+    });
+
+    if (!recipeOfTheDay) {
+      await Recipe.updateMany(
+        { featured: true },
+        { featured: false, lastFeatured: null }
+      );
+
+      const lastWeek = new Date(today);
+      lastWeek.setDate(lastWeek.getDate() - 7);
+
+      const eligibleRecipes = await Recipe.find({
+        $or: [
+          { lastFeatured: null },
+          { lastFeatured: { $lt: lastWeek } }
+        ]
+      });
+
+      if (eligibleRecipes.length > 0) {
+        const randomIndex = Math.floor(Math.random() * eligibleRecipes.length);
+        recipeOfTheDay = eligibleRecipes[randomIndex];
+      } else {
+        const count = await Recipe.countDocuments();
+        const random = Math.floor(Math.random() * count);
+        recipeOfTheDay = await Recipe.findOne().skip(random);
+      }
+
+      recipeOfTheDay.featured = true;
+      recipeOfTheDay.lastFeatured = today;
+      await recipeOfTheDay.save();
+    }
+
+    res.status(200).json({
+      success: true,
+      recipe: recipeOfTheDay
+    });
+
+  } catch (error) {
+    console.error('Error in getRecipeOfTheDay:', error);
+    res.status(500).json({
+      success: false,
+      message: "Error fetching Recipe of the Day",
+      error: error.message
+    });
+  }
+};
```

Comment:
@sanchitc05  you have written the function but it is not exported, how will it work?

Reply:
Let me make the necessary changes